packages/mui-joy/src/Typography/Typography.test.tsx

```diff
@@ -38,12 +41,10 @@ describe('<Typography />', () => {
     expect(container.firstChild).to.have.class(classes.root);
   });
 
-  ['h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'body1', 'body2', 'body3'].forEach((level) => {
+  (['h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'body1', 'body2', 'body3'] as const).forEach((level) => {
     it(`should render ${level} text`, () => {
-      const { container } = render(<Typography level={level}>Hello</Typography>);
-
+      const { container } = render(<Typography level={level as TypographyLevel}>Hello</Typography>);
```

Comment:
```suggestion
      const { container } = render(<Typography level={level}>Hello</Typography>);
```

Is it possible to remove the type casting?

Reply:
> Is it possible to remove the type casting?

When I remove the type casting on level, it throws an error 
`Type '"h1" | "h2" | "h3" | "h4" | "h5" | "h6" | "body3" | "body1" | "body2"' is not assignable to type '"inherit" | keyof TypographySystem | undefined'.`

so I added the options  `"h1" | "h2" | "h3" | "h4" | "h5" | "h6" | "body3" | "body1" | "body2"` to level in TypographyProps like this
`level?:
        | keyof TypographySystem
        | 'inherit'
        | 'h1'
        | 'h2'
        | 'h3'
        | 'h4'
        | 'h5'
        | 'h6'
        | 'body1'
        | 'body2'
        | 'body3';`

The tests are passing, can push this instead?